wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/CommandConverter.java

```diff
@@ -0,0 +1,96 @@
+import org.eclipse.che.api.devfile.server.Constants;
+import org.eclipse.che.api.devfile.server.DevfileFormatException;
+import org.eclipse.che.api.devfile.server.WorkspaceExportException;
+import org.eclipse.che.api.workspace.server.model.impl.CommandImpl;
+
+/**
+ * Helps to convert {@link CommandImpl workspace command} to {@link Command devfile command} and
+ * vice versa.
+ *
+ * @author Sergii Leshchenko
+ */
+public class CommandConverter {
+
+  /**
+   * Converts the specified workspace command to devfile command.
+   *
+   * @param command source workspace command
+   * @return created devfile command based on the specified workspace command
+   * @throws WorkspaceExportException if workspace command does not has specified tool name
+   *     attribute where it should be run
+   */
+  public Command toDevfileCommand(CommandImpl command) throws WorkspaceExportException {
+    Command devCommand = new Command().withName(command.getName());
+    Action action = new Action().withCommand(command.getCommandLine()).withType(command.getType());
+    String workingDir = command.getAttributes().remove(WORKING_DIRECTORY_ATTRIBUTE);
+    if (!isNullOrEmpty(workingDir)) {
+      action.setWorkdir(workingDir);
+    }
+    String toolName = command.getAttributes().remove(Constants.TOOL_NAME_COMMAND_ATTRIBUTE);
+    if (toolName == null) {
+      throw new WorkspaceExportException(
+          format("Command `%s` has no specified tool where it should be run", command.getName()));
+    }
+
+    action.setTool(toolName);
+    action.setType(Constants.EXEC_ACTION_TYPE);
+    devCommand.getActions().add(action);
+    devCommand.setAttributes(command.getAttributes());
+    return devCommand;
+  }
+
+  /**
+   * Converts the specified devfile command to workspace command.
+   *
+   * @param devfileCommand devfile command that should be converted
+   * @return created workspace command based on the specified devfile command
+   */
+  public CommandImpl toWorkspaceCommand(Command devfileCommand) throws DevfileFormatException {
+    if (devfileCommand.getActions().size() != 1) {
+      throw new DevfileFormatException(
+          format("Command `%s` MUST has one and only one action", devfileCommand.getName()));
+    }
+
+    Action commandAction = devfileCommand.getActions().get(0);
+
+    return toWorkspaceCommand(devfileCommand, commandAction);
+  }
+
+  private CommandImpl toWorkspaceCommand(Command devCommand, Action commandAction) {
+    CommandImpl command = new CommandImpl();
+    command.setName(devCommand.getName());
+    command.setType(commandAction.getType());
+    command.setCommandLine(commandAction.getCommand());
+
+    if (commandAction.getWorkdir() != null) {
+      command.getAttributes().put(WORKING_DIRECTORY_ATTRIBUTE, commandAction.getWorkdir());
+    }
+
+    command.getAttributes().put(Constants.TOOL_NAME_COMMAND_ATTRIBUTE, commandAction.getTool());
+
+    if (devCommand.getAttributes() != null) {
```

Comment:
Actually, we should not have this check since `devCommand` is POJO and we suppose that lists/sets/maps are not nullable in POJOs. Will remove this check.